libs/librepcb/projecteditor/boardeditor/fsm/bes_drawtrace.cpp

```diff
@@ -918,23 +916,27 @@ void BES_DrawTrace::updateNetpointPositions() noexcept {
     // find anchor under cursor
     GraphicsLayer* layer = mPositioningNetPoint1->getLayerOfLines();
     Q_ASSERT(layer);
-    if (BI_Via* via =
-            findVia(board, mCursorPos, mCurrentNetSignal, {mTempVia})) {
+    NetSignal* netsignal = &mCurrentNetSegment->getNetSignal();
+    // NOTE(5n8ke): netsignal must not be nullptr, since a connection should
+    // only be made to the current NetSignal
+    Q_ASSERT(netsignal);
```

Comment:
`netsignal` can't be `nullptr` since it was constructed by taking the address of an object, so no check is needed:

```suggestion
    NetSignal* netsignal = &mCurrentNetSegment->getNetSignal();
```

But actually, one more time, simply don't convert the reference to a pointer here, then it's obvious that it can't be `nullptr` :smiley: :

```suggestion
    NetSignal& netsignal = mCurrentNetSegment->getNetSignal();
```

Then use `&netsignal` where you really need a pointer.

Reply:
I chose to use a pointer here since all the following find* take a pointer as the netsignal parameter.
The check is not needed, but I added the asserts as a sanity check to make it explicit, since a null pointer was previously possible with mCurrentNetSignal.

Reply:
Hmm okay - to be honest, I don't like it (IMO a reference would be much cleaner) but this time I'll turn a blind eye :wink: